homeassistant/components/honeywell/climate.py

```diff
@@ -384,12 +384,12 @@ async def async_turn_aux_heat_off(self) -> None:
 
     async def async_update(self) -> None:
         """Get the latest state from the service."""
+        if not self._attr_available:
+            self._attr_available = await self._data.retry_login()
         try:
             await self._device.refresh()
```

Comment:
Should we still try a refresh if the login above returns False?

Reply:
Login function returns false when the web site is down.  This site uses session keys, so if we get disconnected, we need to try to get new keys.  Is there a better way to retry connecting? 
I definitely can tell in the retry_login() function if it's a connection or password issue... should I raise something to stop trying if it's a password problem?  I don't believe raising ConfigEntryNotReady or ConfigEntryAuthFailed will at this level (right?)

Reply:
We can start a reauth flow with `config_entry.async_start_reauth`. But the entity update will still be scheduled then.

Another alternative is to reload the integration and let the config entry setup raise `ConfigEntryAuthFailed`.

Reply:
Reload the integration sounds like a better plan - that would start the retry process depending on the type of error.  
Should I still set the available to False then reload?

Reply:
What is the reload path from inside an entity?
await self.hass.config_entries.async_reload(self.entry.entry_id)

Reply:
Yes, unless we're already unavailable it's good to set available to False so that the state is updated correctly for this update.